### PR TITLE
Issue 44 Build Initial Wikipedia Knowledge Base

### LINKED ISSUE #44: Build Initial Wikipedia Knowledge Base
As a developer, I want to collect reliable information about selected destinations so that the AI assistant can ground its answers in real-world facts.

---

**Description**
Gather travel-related content (activities, attractions, restaurants, cultural notes, local tips) for 2 initial cities (e.g., **Kyiv**, **Lviv**) to serve as the factual base for later embedding and retrieval.

---

**Goals & Tasks**

* [ ] Select 2 target cities (team decision or mentor input)
* [ ] Collect 20 sources per city:

  * Wikipedia articles*
* [ ] Extract relevant sections:

  * Top sights
  * Activities for families, foodies, etc.
  * Seasonal notes
* [ ] Save each document in plain `.txt` or `.md` format under `/data/raw/`
* [ ] Create metadata file with source info:
  `/data/metadata.csv`: columns = `city,source_type,url,summary`

---

**Expected Output**

* Folder structure:

  ```

  /data/
    └── raw/
        ├── kyiv_01.txt
        ├── lviv_01.txt
         ....
    └── metadata.csv
    └── attractions_names_list.csv
  ```
* Summary in `/docs/knowledge-base-summary.md`

### PR DESCRIPTION
**Pull request by sskyisthelimit**
## Description
Gather content about attractions for Kyiv and Lviv to serve as the factual base for later embedding and retrieval.

## Changes
- Selected target cities: **Kyiv**, **Lviv**  
- Collected 20 Wikipedia sources per city  
- Saved each document as `.txt` under `/data/raw/`  
- Created `/data/metadata.csv` (`city,source_type,url,summary`)  
- Created `/data/attractions_names_list.csv`  
- Added summary in `/docs/knowledge-base-summary.md`  

## Folder Structure
/data/
└── raw/
├── kyiv_....txt … kyiv_....txt
├── lviv_...txt … lviv_...txt
└── metadata.csv
└── attractions_names_list.csv

/docs/
└── knowledge-base-summary.md

## Known Problems / Decisions
1. Non-Latin words are deleted.  
2. Section titles retained as `== Title ==` for chunking.  
3. Line breaks (`\n`) preserved.  
4. Images are not extracted.  
5. Content from “See also” onward is omitted.

Closes #44

### CODE REVIEW

**Comment by Roman-Terendiy** [comment]
Everything looks clean and organized for the next steps. Nice work!

File: app/retrieval/parsing/attractions_wiki_parser.py
[the remark this answers is not in the document]
Comment (sskyisthelimit) [reply app/retrieval/parsing/attractions_wiki_parser.py:1]: added app/cli/attractions_cli.py

[the remark this answers is not in the document]
Comment (sskyisthelimit) [reply app/retrieval/parsing/attractions_wiki_parser.py:1]: added tests/test_text_cleaning.py, all tests pass

File: tests/test_text_cleaning.py [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+        * Link 1
+        * Link 2
+        
+        More content.
+        """
+        result = self.parser._remove_see_also_section(text)
+        self.assertNotIn("== See also ==", result)
+        self.assertNotIn("Link 1", result)
+        self.assertIn("Some content here", result)
+        self.assertIn("More content", result)
```
Comment (mehalyna) [inline tests/test_text_cleaning.py:149]: add cases where "See also" is embedded in other text or is formatted differently (e.g., extra whitespace, tabs)

File: tests/test_text_cleaning.py [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+        self.assertNotIn("*** Sub-sub-item", result)
+        self.assertIn("Item 1", result)
+        self.assertIn("Sub-item 1", result)
+    
+    def test_cleanup_whitespace(self):
+        """Test whitespace cleanup."""
+        text = "Multiple    spaces    here.\n\n\nMultiple\n\n\nnewlines."
+        result = self.parser._cleanup_whitespace(text)
+        self.assertNotIn("    ", result)  # No multiple spaces
+        self.assertIn("Multiple spaces here", result)
```
Comment (mehalyna) [inline tests/test_text_cleaning.py:297]: also test for tabs and irregular whitespace characters

File: tests/test_text_cleaning.py [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+        self.assertNotIn("* List item", result)
+        self.assertNotIn("    ", result)  # No multiple spaces
+        
+        # Check that content is preserved
+        self.assertIn("Hello", result)
+        self.assertIn("there", result)
+        self.assertIn("Bold", result)
+        self.assertIn("italic", result)
+        self.assertIn("List item", result)
+        self.assertIn("Sub-item", result)
```
Comment (mehalyna) [inline tests/test_text_cleaning.py:336]: add assertions for expected cleaned output, not just the absence of unwanted patterns

File: tests/test_text_cleaning.py [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+    
+    def test_empty_inputs(self):
+        """Test handling of empty inputs."""
+        empty_inputs = ["", None, "   ", "\n\n\n"]
+        
+        for empty_input in empty_inputs:
+            with self.subTest(empty_input=empty_input):
+                if empty_input is not None:
+                    result = self.parser._apply_text_cleaning_regex(empty_input)
+                    self.assertIsInstance(result, str)
```
Comment (mehalyna) [inline tests/test_text_cleaning.py:387]: add explicit assertions for expected output (e.g., empty string, None)

File: tests/test_text_cleaning.py [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+            ("({{template}})", ""),  # Template without params
+            ("({{template|}})", ""),  # Template with empty param
+            ("({{template|param1|param2}})", ""),  # Multiple params
+        ]
+        
+        for input_text, expected in test_cases:
+            with self.subTest(input_text=input_text):
+                result = self.parser._remove_template_parentheses(input_text)
+                self.assertEqual(result, expected)
+
```
Comment (mehalyna) [inline tests/test_text_cleaning.py:410]: - For template edge cases, add more malformed templates and parenthesis combinations.
- Add tests for extremely long strings or very large templates to check performance and memory handling.

File: tests/test_text_cleaning.py
Comment (mehalyna) [inline tests/test_text_cleaning.py:1]: - Use more descriptive test method names, e.g., `test_remove_non_latin_words_with_preserve_mixed_enabled`.
- Add coverage for exceptions and ensure that failing cases are handled gracefully.
- Consider breaking up very long test files into multiple files by class/module if the test suite grows.
- Document any custom test utilities, fixtures, or helpers at the top of the file.
- Add tests for public methods that use these private cleaning functions if any exist.

File: tests/test_text_cleaning.py [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (mehalyna) [inline tests/test_text_cleaning.py:413]: - Use unittest.main(verbosity=2) for more informative output during test runs.
- If you gonna use pytest, remove the main guard and let pytest discover your tests.
